### PR TITLE
webhooks: add signature check

### PR DESCRIPTION
**Pull request by vxio**
Note: branched off of #97 

**Changes**
1. Add signature check to ParseEvent
2. Update test in `examples/webhooks_test.go`

### CODE REVIEW

File: pkg/mhooks/event.go [hunk 1]
```diff
@@ -2,14 +2,24 @@ package mhooks
...
 	"encoding/json"
+	"errors"
 	"fmt"
-	"io"
+	"net/http"
 	"time"
 )
 
-func NewEvent(requestBody io.Reader) (*Event, error) {
+func ParseEvent(r *http.Request, secret string) (*Event, error) {
```
Comment (vxio) [inline pkg/mhooks/event.go:11]: Originally, I had `CheckSignature` as a separate method that users could call before `ParseEvent`. But this way, we force users to check the signature by just providing the secret. Thoughts?

Comment (adamdecaf) [reply pkg/mhooks/event.go:11]: Requiring the signature is checked seems fine to me. People can still unmarshal `Event` if they really want to anyway.

File: examples/webhooks/webhooks_test.go [hunk 2]
```diff
@@ -74,14 +80,20 @@ func TestNewEvent(t *testing.T) {
...
-				CreatedOn: time.Now().UTC(),
+				CreatedOn: createdOn,
 			}
 
 			var body bytes.Buffer
 			err = json.NewEncoder(&body).Encode(event)
 			require.NoError(t, err)
 
 			req := httptest.NewRequest(http.MethodPost, "/my-awesome-webhook-url", &body)
+			req.Header.Set("x-timestamp", timestamp)
```
Comment (InfernoJJ) [inline examples/webhooks/webhooks_test.go:91]: Why not break out the header testing in another test so its easier to document and present how it all comes together? I worry this test of parsing events is going to get way to long and become unreadable.

**Review by InfernoJJ — APPROVED** [review]
Good for a start. I still worried about that test becoming un-readable as we add events.

**Review by InfernoJJ — CHANGES_REQUESTED** [review]
We need examples to match what a production result could look like.

**Comment by vxio** [comment]
> We need examples to match what a production result could look like.

I'll fix this up in a separate PR so it's easier to review; I'm going to add both unit tests and actual examples (https://go.dev/blog/examples)

**Review by InfernoJJ — APPROVED** [review]

**Pull request merged**